mip/driver_stm32.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+struct mip_driver_stm32 {
+  // MDC clock divider. MDC clock is derived from HCLK, must not exceed 2.5MHz
+  //    HCLK range    DIVIDER    mdc_cr VALUE
+  //    -------------------------------------
+  //    60-100 MHz    HCLK/42        0
+  //    100-150 MHz   HCLK/62        1
+  //    20-35 MHz     HCLK/16        2
+  //    35-60 MHz     HCLK/26        3
+  //    150-216 MHz   HCLK/102       4  <-- value for Nucleo-F* on max speed
+  //    216-310 MHz   HCLK/124       5
```

Comment:
These values are just probably a coincidence for an MCU with 216 max speed, actual values are taken using a similar approach to what we've taken. (Though they seem to use a fixed guard instead of a percentage)

Reply:
The main motivation for this was: if user needs to set a custom value, they would need to read and understand the code of the driver, and what exactly define to set - which is not trivial.
This approach makes it clear, which tunables are there and what are the possible values.

Reply:
Totally agree, and the -1 option is even cooler, great idea !

Reply:
Done